### ISSUE TITLE
LevelDB tmp directory

### ISSUE THREAD

**Issue by laxatives** [pelias/openstreetmap#257]
My understanding is the leveldb directory is used to dedupe already imported addresses. Its approaching 100GB for my import -- is it safe for me to delete after the import is done? I haven't specified whether deduplication should be run in the config (I assumed it was disabled by default) -- can I delete it now? 

What is the behavior if I drop my current elasticsearch `pelias` index and start a new import? Will I be missing documents if I didn't also clear the leveldb directory? 

What if I clear the leveldb directory and rerun an importer -- will I have duplicate documents in this case?

**Comment by missinglink** [comment]
the `leveldb` directory is used to store the lat/lon of OSM nodes, it's not used for address deduplication.

for example imagine the PBF file contains:
```
node[100]: 1,1
node[101]: 2,2
node[102]: 3,3
way[99]: [ 100, 101, 102 ]
```

we store the lat/lon of nodes `100,101,102` in leveldb and then when we get to way `99` we need to look up the node coordinates in order to build the linestring: `[ [ 1,1 ], [ 2, 2 ], [ 3, 3 ] ]`

the reason it's 100GB is because the planet file has a [lot of nodes](http://www.openstreetmap.org/stats/data_stats.html) in it ~3Billion

once the import is done it is safe to delete this directory.

if you start a new importer then it should also clear the leveldb directory when it starts, if it doesn't to that then please let me know and I'll fix it.

you will never get duplicates of OSM data because the index key is the OSM id, so if you were to index a document twice it's value would simply change to the last value you set it to.

**Comment by laxatives** [comment]
Ah, gotcha thanks.
